NativeApp/src/screens/Logout.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react';
+import { View, Alert } from 'react-native';
+import { PropTypes as PT } from 'prop-types';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import { userLogout } from '../utilities/currentUser';
+
+export default class LogoutScreen extends Component {
+  static navigationOptions = {
+    header: null,
+    title: 'Logout',
+    tabBarIcon: ({ tintColor }) => <Icon name="sign-out" size={25} color={tintColor} />,
+  };
+
+  static propTypes = {
+    navigation: PT.shape({
+      navigate: PT.func,
+    }).isRequired,
+  }
+
+  componentDidMount() {
+    const { navigation } = this.props;
+
+    this.sub = navigation.addListener('didFocus', () => {
+      Alert.alert(
+        'Logout?',
+        'Are you sure you want to logout?',
+        [
+          { text: 'Yes', onPress: () => this.handleLogout() },
+          { text: 'No', onPress: () => navigation.goBack() },
+        ],
+        { cancelable: false },
+      );
+    });
+  }
+
+  componentWillUnmount() {
+    this.sub.remove();
+  }
+
+  handleLogout = () => {
+    const { navigation } = this.props;
+    userLogout()
+      .then(navigation.navigate('Auth'));
```

Comment:
I'm not 100% on this, but I think this might be firing immediately rather than after a successful `userLogout`. I would have thought it needed to be `.then(() => navigation.navigate('Auth'))`

Reply:
In my head, as soon as `handleLogout` is called, it's evaluating the `navigation.navigate('Auth')` call immediately and passing that to `then`, whereas we want to pass a function that performs the navigation.

It'd be interesting to stick some logging in and see the order in which things are firing, but if you don't have time I say include `.then(() => navigation.navigate('Auth'))` just to be safe 👍

Reply:
Actually @bgribz your right. The navigation was being called first. Good spot! 👍